content/working_with/official_plugins/Orchestration/helm.md

```diff
@@ -0,0 +1,465 @@
+              user:
+                client-certificate-data: { get_input: kubernetes-admin_client_certificate_data }
+                client-key-data:{ get_input: kubernetes-admin_client_key_data }
+      resource_config:
+        name: "myrelease"
+        chart: { concat: [ { get_input: repo_name },'/', { get_input: chart_name } ] }
+    relationships:
+      - target: helm_install
+        type: cloudify.helm.relationships.run_on_host
+      - target: repo
+        type: cloudify.relationships.depends_on
+
+{{< /highlight >}}
+
+**Example 3:**
+
+This is an example for authentication with kubeconfig file path(the file is in the blueprint archive):
+
+{{< highlight  yaml  >}}
+
+node_templates:
+  
+  release:
+    type: cloudify.nodes.helm.Release
+    properties:
+      client_config:
+        configuration:
+          blueprint_file_name: path/to/kubeconfig
+      resource_config:
+        name: "myrelease"
+        chart: { concat: [ { get_input: repo_name },'/', { get_input: chart_name } ] }
+    relationships:
+      - target: helm_install
+        type: cloudify.helm.relationships.run_on_host
+      - target: repo
+        type: cloudify.relationships.depends_on
+
+{{< /highlight >}}
+
+# Node Types
+
+## cloudify.nodes.helm.Binary
+
+This node type responsible for installing helm (move the given binary to the default location).
+
+Moreover, in order to allow different environment for every binary, this node will create 3 temporary folders for: cache, data and configuration files of Helm. It will save those paths in runtime properties.
+
+Actually, those paths are going to override HELM_CACHE_HOME,HELM_CONFIG_HOME and HELM_DATA_HOME environment variables in each helm command we will execute.
+
+### Properties:
+  * `helm_config` - Represents Helm configuration.
+    
+    *type:* cloudify.types.helm.HelmConfig
+    
+    Currently contains `executable_path` with default value of '/usr/bin/helm'.
+  * `use_existing_resource` - If true, use an existing helm installation rather than installing it.
+    
+    *type:* boolean
+    
+    *default:* False
+  * `installation_source` - Location to download the Helm installation from. Ignored if `use_existing_resource` is true.
+    
+    *type:* string
+    
+    *default:* ''
+    
+    You can see helm releases [here](https://github.com/helm/helm/releases) please use helm 3.X.X version.
+
+Helm plugin uses `curl` on  `installation_source` and unzip it, then move it to `executable_path` or to default location if `executable_path` is not provided.
+
+### Example:
+
+{{< highlight  yaml  >}}
+
+inputs:
+
+  helm_executable:
+    description: >
+      Helm binary path.
+    type: string
+    default: '/tmp/helm'
+
+  helm_installation_source:
+    description: >
+      Helm download link.
+    type: string
+    default: 'https://get.helm.sh/helm-v3.3.1-linux-amd64.tar.gz'
+
+dsl_definitions:
+
+  helm_config: &helm_config
+    executable_path: { get_input: helm_executable }
```

Comment:
I would keep here just the node template. And be verbose in the node type. No  good to use inputs and dsl_definitions in the example here.

Reply:
Done